homeassistant/components/websocket_api/commands.py

```diff
@@ -555,3 +536,40 @@ async def handle_fire_event(
 
     hass.bus.async_fire(msg["event_type"], msg.get("event_data"), context=context)
     connection.send_result(msg["id"], {"context": context})
+
+
+@decorators.websocket_command(
+    {
+        vol.Required("type"): "validate_config",
+        vol.Optional("trigger"): cv.match_all,
+        vol.Optional("condition"): cv.match_all,
+        vol.Optional("action"): cv.match_all,
+    }
+)
+@decorators.async_response
+async def handle_validate_config(
+    hass: HomeAssistant, connection: ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Handle validate config command."""
+    # Circular dep
+    # pylint: disable=import-outside-toplevel
+    from homeassistant.helpers import condition, script, trigger
+
+    result = {}
+
+    for key, schema, validator in (
+        ("trigger", cv.TRIGGER_SCHEMA, trigger.async_validate_trigger_config),
+        ("condition", cv.CONDITION_SCHEMA, condition.async_validate_condition_config),
+        ("action", cv.SCRIPT_SCHEMA, script.async_validate_actions_config),
+    ):
+        if key not in msg:
+            continue
+
+        try:
+            await validator(hass, schema(msg[key]))  # type: ignore
```

Comment:
Please add the ignore error code when ignoring a type error.

Reply:
https://github.com/home-assistant/core/pull/67311